[13.0] [IMP] report_qweb_signer: allowed reports on certificate

ThomasBinsfeld opened the pull request:
Add a new 'Allowed reports' field on the certificate. Only reports listed in this field can be signed. No report means all reports are allowed.

ThomasBinsfeld committed:
[IMP] report_qweb_signer: allowed reports on certificate
Add a new 'Allowed reports' field on the certificate. Only reports listed in this field can be signed. No report means all reports are allowed.

pedrobaeza commented:
I suppose you need this because not all the reports of the selected model should be signed. Why not taking the occasion to add the whole configuration at report level (if to be signed or not, the domain, etc...)?

ThomasBinsfeld commented:
> I suppose you need this because not the reports of the selected model should be signed. 

Yes

> Why not taking the occasion to add the whole configuration at report level (if to be signed or not, the domain, etc...)?

Not sure there is a real added value to do that. Wyt?

pedrobaeza commented:
Well, we decentralize the configuration per document, and avoid side effects if a new report is added in the model and it gets automatically signed because we haven't excluded it.

chienandalu reviewed (approved):
Aside from @pedrobaeza suggestion I think is ok :+1:

chienandalu commented:
Maybe a comment can be added to the ROADMAP so we can move on with this change.

chienandalu commented:
@ThomasBinsfeld I think this is a good addition to the module. Only a ROADMAP map with @pedrobaeza was left, I think. Will you finish it?

ThomasBinsfeld committed:
[ADD] report_qweb_signer: add  pedrobaeza idea in ROADMAP

ThomasBinsfeld commented:
> @ThomasBinsfeld I think this is a good addition to the module. Only a ROADMAP map with @pedrobaeza was left, I think. Will you finish it?

Thanks for the review @chienandalu 

I rebased the branch and solved the conflicts. This PR is in production for more than a year and is ready to be merged IMO.

I added @pedrobaeza 's refactoring idea in the ROADMAP since i have neither the time nor the intention to implement it.

pedrobaeza commented:
Hi, Thomas, you should add it in readme/ROADMAP.rst, not in README.rst

ThomasBinsfeld commented:
> Hi, Thomas, you should add it in readme/ROADMAP.rst, not in README.rst

Fixed, thanks

pedrobaeza commented:
/ocabot merge major

chienandalu commented:
Will you forward it to latter versions?

ThomasBinsfeld commented:
> Will you forward it to latter versions?

Here it is https://github.com/OCA/reporting-engine/pull/668
